metabase/metabase
# Links to metabase URLs should open in the same window in link cards

Partially fixes https://github.com/metabase/metabase/issues/30891

### Description

External link cards currently always opens in new tabs. When clicking on a link to metabase urls, such as `https://example.com/question/1-some-question`, it opens a new tab instead of opening in the same window. This makes the URL break out of embedding as well.

This PR makes it so links with the same origin or same as the site url open in the same window.

### Caveat

This also causes all relative links to open in the same tab. This would make relative links to questions and dashboards open correctly, but invalid relative links such as `example.com`, `www.example.com`, `foobar` will remain broken. An open question is how should we handle invalid relative links in external link cards.

### How to verify

- Create a new link card in a dashboard.
- Enter the absolute url to a metabase question or dashboard, e.g. `https://localhost:3000/dashboard/1-example`
- Click on the card, and it should open in the same page. Go back to the dashboard.
- Next, edit the card and enter a relative url to a metabase question or dashboard, e.g. `dashboard/1-example`
- Click on the card, and it should also open in the same page

### Checklist
